src/common/config/metadata/value.go

```diff
@@ -0,0 +1,143 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"errors"
+
+	"github.com/goharbor/harbor/src/common/utils/log"
+)
+
+var (
+	// ErrNotDefined ...
+	ErrNotDefined = errors.New("configure item is not defined in metadata")
+	// ErrTypeNotMatch ...
+	ErrTypeNotMatch = errors.New("the required value doesn't matched with metadata defined")
+	// ErrInvalidData ...
+	ErrInvalidData = errors.New("the data provided is invalid")
+	// ErrValueNotSet ...
+	ErrValueNotSet = errors.New("the configure value is not set")
+)
+
+// ConfigureValue - struct to hold a actual value, also include the key of config metadata.
+type ConfigureValue struct {
+	Key   string `json:"key,omitempty"`
+	Value string `json:"value,omitempty"`
+}
+
+// NewConfigureValue ...
+func NewConfigureValue(key, value string) *ConfigureValue {
+	result := &ConfigureValue{}
+	if _, ok := Instance().GetMetaData(key); ok {
+		err := result.Set(key, value)
+		if err == nil {
+			return result
+		}
+		log.Errorf("Failed to set key:%v, value:%v, error %v", key, value, err)
+	} else {
+		log.Errorf("Failed to create a configure value without metadata, key %v, value %v", key, value)
+	}
+	return result
+}
+
+// GetString - Get the string value of current configure
+func (c *ConfigureValue) GetString() string {
+	// Any type has the string value
+	if _, ok := Instance().GetMetaData(c.Key); ok {
+		return c.Value
+	}
+	return ""
+}
+
+// GetKey ...
+func (c *ConfigureValue) GetKey() string {
+	return c.Key
+}
+
+// GetInt - return the int value of current value
+func (c *ConfigureValue) GetInt() int {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if intValue, suc := item.ItemType.get(c.Value).(int); suc {
+			return intValue
+		}
+	}
+	log.Errorf("The current value can not convert to integer, %+v", c)
+	return 0
+}
+
+// GetInt64 - return the int64 value of current value
+func (c *ConfigureValue) GetInt64() int64 {
+	if item, ok := Instance().GetMetaData(c.Key); ok {
+		if int64Value, suc := item.ItemType.get(c.Value).(int64); suc {
+			return int64Value
+		}
+	}
+	log.Errorf("Current value can not convert to int64, %+v", c)
+	return 0
+}
+
+// GetBool - return the bool value of current setting
+func (c *ConfigureValue) GetBool() bool {
```

Comment:
return 0 or false is as designed?

Reply:
Yes

Reply:
the default value could be discussed, but let't merge the PR first.